cmd/main.go

```diff
@@ -110,10 +111,28 @@ func main() {
 	nodeName := os.Getenv("NODE_NAME")
 	node, err := k8sClientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
 	if err != nil {
-		logger.Info("Error retrieving the Node from the index for a given node. Error :", zap.Error(err))
 		if errors.IsNotFound(err) {
 			runtimeu.HandleError(fmt.Errorf("node '%s' no longer exist in the cluster", nodeName))
 		}
+		// Do multiple retries if there is connection error.
+		logger.Info("Retrying connection to node")
+		errRetry := nodeupdater.ErrorRetry(logger, func() (error, bool) {
+			node, err = k8sClientset.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
+			if err != nil {
```

Comment:
I think, we can simply retry for all the error that would be good so that this code will look simple although I don't see any point to categories errors here